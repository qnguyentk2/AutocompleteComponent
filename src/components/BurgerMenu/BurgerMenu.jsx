import React, { useContext } from 'react';
import { slide as Menu } from 'react-burger-menu';
import  FontAwesome from 'react-fontawesome';
// import classNames from 'classnames';
import { MENU_ITEMS } from '@const/common';
import GlobalContext from '@contexts/GlobalContext';
import BurgerMenuStyles from './BurgerMenu.scss';

const burgerStyles = {
  bmBurgerButton: {
    position: 'relative',
    width: '26px',
    height: '20px',
    top: '5px'
  },
  bmBurgerBars: {
    background: '#fff'
  },
  bmBurgerBarsHover: {
    background: '#a90000'
  },
  bmCrossButton: {
    height: '24px',
    width: '24px'
  },
  bmCross: {
    background: '#bdc3c7'
  },
  bmMenuWrap: {
    position: 'fixed',
    height: '100%',
    top: 0
  },
  bmMenu: {
    background: 'rgb(98, 101, 113)',
    padding: '2.5em 1.5em 0',
    fontSize: '1.15em'
  },
  bmMorphShape: {
    fill: '#373a47'
  },
  bmItemList: {
    color: '#b8b7ad',
    padding: '0.8em',
    display: 'flex',
    flexDirection: 'column'
  },
  bmItem: {
    display: 'inline-flex',
    padding: '10px',
    fontSize: '16px',
    color: 'aliceblue',
    textDecoration: 'unset',
  },
  bmOverlay: {
    background: 'rgba(14, 1, 1, 0.58)',
    top: 0,
    left: 0
  }
}

MENU_ITEMS.reverse();

const BurgerMenu = () => {
  const { handleDetectBurgerMenu } = useContext(GlobalContext);

  return (
    <div>
      <Menu onStateChange={handleDetectBurgerMenu} pageWrapId="app" right styles={{...burgerStyles}} outerContainerId="app">
        {
          MENU_ITEMS.map(item => (
            <div className={BurgerMenuStyles.itemWrapper}>
              {
                item.icon && (
                  <div className={BurgerMenuStyles.iconWrapper}>
                    <FontAwesome 
                      className="super-crazy-colors"
                      name={item.icon}
                      size="1.5x"
                      // spin
                      style={{ textShadow: '0 1px 0 rgba(0, 0, 0, 0.1)', padding: '9px' }}
                    />
                  </div>
                )
              }
              <div className={BurgerMenuStyles.menuItem}>
                <a key={item.label} id={item.label}>
                  {item.label.toLocaleUpperCase()}
                </a>
              </div>
            </div>
          ))
        }
      </Menu>
    </div>
  )
}

export default BurgerMenu;

BurgerMenu.propTypes = {
  // showMenu: PropTypes.func,
};

BurgerMenu.defaultProps = {
  // showMenu: () => {},
};