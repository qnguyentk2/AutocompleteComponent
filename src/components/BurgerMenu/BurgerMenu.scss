.iconWrapper {
  text-align: center;
  width: 20%;
}
.menuItem {
  padding: 10px;
  a {
    text-decoration: none;
    color: aliceblue;
  }
}