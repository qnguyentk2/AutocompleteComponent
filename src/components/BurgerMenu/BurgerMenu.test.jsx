/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import BurgerMenu from './BurgerMenu';

const setup = () => {
  const wrapper = shallow(<BurgerMenu />);

  return { wrapper };
}

describe('Test BurgerMenu', () => {
  test('renders BurgerMenu without crashing', () => {
    const { wrapper } = setup();
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
})