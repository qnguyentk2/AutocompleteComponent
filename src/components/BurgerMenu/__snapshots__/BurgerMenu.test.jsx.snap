// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test BurgerMenu renders BurgerMenu without crashing 1`] = `
<div>
  <Menu
    bodyClassName=""
    burgerBarClassName=""
    burgerButtonClassName=""
    className=""
    crossButtonClassName=""
    crossClassName=""
    disableAutoFocus={false}
    disableCloseOnEsc={false}
    htmlClassName=""
    id=""
    itemClassName=""
    itemListClassName=""
    menuClassName=""
    morphShapeClassName=""
    noOverlay={false}
    noTransition={false}
    onIconHoverChange={[Function]}
    onStateChange={[Function]}
    outerContainerId="app"
    overlayClassName=""
    pageWrapId="app"
    right={true}
    styles={
      Object {
        "bmBurgerBars": Object {
          "background": "#fff",
        },
        "bmBurgerBarsHover": Object {
          "background": "#a90000",
        },
        "bmBurgerButton": Object {
          "height": "20px",
          "position": "relative",
          "top": "5px",
          "width": "26px",
        },
        "bmCross": Object {
          "background": "#bdc3c7",
        },
        "bmCrossButton": Object {
          "height": "24px",
          "width": "24px",
        },
        "bmItem": Object {
          "color": "aliceblue",
          "display": "inline-flex",
          "fontSize": "16px",
          "padding": "10px",
          "textDecoration": "unset",
        },
        "bmItemList": Object {
          "color": "#b8b7ad",
          "display": "flex",
          "flexDirection": "column",
          "padding": "0.8em",
        },
        "bmMenu": Object {
          "background": "rgb(98, 101, 113)",
          "fontSize": "1.15em",
          "padding": "2.5em 1.5em 0",
        },
        "bmMenuWrap": Object {
          "height": "100%",
          "position": "fixed",
          "top": 0,
        },
        "bmMorphShape": Object {
          "fill": "#373a47",
        },
        "bmOverlay": Object {
          "background": "rgba(14, 1, 1, 0.58)",
          "left": 0,
          "top": 0,
        },
      }
    }
    width={300}
  >
    <div>
      <div
        className="iconWrapper"
      >
        <FontAwesome
          className="super-crazy-colors"
          name="sign-in"
          size="1.5x"
          style={
            Object {
              "padding": "9px",
              "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
            }
          }
        />
      </div>
      <div
        className="menuItem"
      >
        <a
          id="Log in"
          key="Log in"
        >
          LOG IN
        </a>
      </div>
    </div>
    <div>
      <div
        className="iconWrapper"
      >
        <FontAwesome
          className="super-crazy-colors"
          name="user-plus"
          size="1.5x"
          style={
            Object {
              "padding": "9px",
              "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
            }
          }
        />
      </div>
      <div
        className="menuItem"
      >
        <a
          id="Sign up"
          key="Sign up"
        >
          SIGN UP
        </a>
      </div>
    </div>
    <div>
      <div
        className="iconWrapper"
      >
        <FontAwesome
          className="super-crazy-colors"
          name="info"
          size="1.5x"
          style={
            Object {
              "padding": "9px",
              "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
            }
          }
        />
      </div>
      <div
        className="menuItem"
      >
        <a
          id="Help"
          key="Help"
        >
          HELP
        </a>
      </div>
    </div>
    <div>
      <div
        className="iconWrapper"
      >
        <FontAwesome
          className="super-crazy-colors"
          name="crosshairs"
          size="1.5x"
          style={
            Object {
              "padding": "9px",
              "textShadow": "0 1px 0 rgba(0, 0, 0, 0.1)",
            }
          }
        />
      </div>
      <div
        className="menuItem"
      >
        <a
          id="Become a host"
          key="Become a host"
        >
          BECOME A HOST
        </a>
      </div>
    </div>
  </Menu>
</div>
`;
